lib/src/main/java/graphql/nadel/validation/hydration/NadelHydrationValidation.kt

```diff
@@ -511,6 +533,43 @@ class NadelHydrationValidation internal constructor(
         return NadelBatchHydrationMatchStrategy.MatchObjectIdentifiers(matchObjectIdentifiers).asInterimSuccess()
     }
 
+    private fun getSourceIdCastStrategy(
+        resultIdField: GraphQLFieldDefinition,
+        sourceIdField: GraphQLFieldDefinition,
+    ): NadelObjectIdentifierCastingStrategy {
+        fun isString(type: GraphQLUnmodifiedType) = (type == GraphQLID || type == GraphQLString)
+        val sourceIdType = sourceIdField.type.unwrapAll()
+        val resultIdType = resultIdField.type.unwrapAll()
+        return if (!isString(sourceIdType) && isString(resultIdType)) {
+            NadelObjectIdentifierCastingStrategy.TO_STRING
+        } else {
+            NadelObjectIdentifierCastingStrategy.NO_CAST
+        }
+    }
+
+    context(NadelValidationContext)
+    private fun getResultIdFieldOrNull(
+        hydration: NadelHydrationDefinition,
+        resultIdFieldName: String,
+    ): GraphQLFieldDefinition? {
```

Comment:
This _should_ be non-null but it's a bit of a can of worms I'm not willing to deal with right now.